File: bench/words.lua
local many = 2000
local rounds = 20

local found = 0
for round = 0, rounds - 1 do
    local pieces = {}
    for i = 0, many - 1 do
        pieces[i + 1] = string.format("item %d of %d in round %d", i, many,
                                      round)
    end
    local whole = table.concat(pieces, ",")
    if string.find(whole, "item 1999 ", 1, true) then
        found = found + 1
    end
    local back = {}
    local from = 1
    while true do
        local at = string.find(whole, ",", from, true)
        if at == nil then
            back[#back + 1] = string.sub(whole, from)
            break
        end
        back[#back + 1] = string.sub(whole, from, at - 1)
        from = at + 1
    end
    if #back ~= many then return end
    for _, one in ipairs(back) do
        if string.sub(one, 1, 7) == "item 1 " then
            found = found + 1
        end
    end
    found = found + (#whole % 7)
end

print(string.format("words %d", found))

File: bench/words.cpp
#include <cstdio>
#include <string>
#include <vector>

int main(void) {
    const int many = 2000;
    const int rounds = 20;
    long long found = 0;
    for (int round = 0; round < rounds; round++) {
        std::vector<std::string> pieces;
        pieces.reserve(many);
        for (int i = 0; i < many; i++) {
            pieces.push_back("item " + std::to_string(i) + " of " +
                             std::to_string(many) + " in round " +
                             std::to_string(round));
        }
        std::string whole;
        for (size_t i = 0; i < pieces.size(); i++) {
            if (i > 0) {
                whole += ",";
            }
            whole += pieces[i];
        }
        if (whole.find("item 1999 ") != std::string::npos) {
            found++;
        }
        std::vector<std::string> back;
        size_t from = 0;
        while (true) {
            size_t at = whole.find(',', from);
            if (at == std::string::npos) {
                back.push_back(whole.substr(from));
                break;
            }
            back.push_back(whole.substr(from, at - from));
            from = at + 1;
        }
        if ((int)back.size() != many) {
            return 1;
        }
        for (const std::string &one : back) {
            if (one.rfind("item 1 ", 0) == 0) {
                found++;
            }
        }
        found += (long long)(whole.size() % 7);
    }
    printf("words %lld\n", found);
    return 0;
}
